src/raytrace/raytrace_gpu.jl

```diff
@@ -0,0 +1,172 @@
+function traverse_rays_nodes!(bvh, points, directions, src, dst, num_src, dst_offsets, level, options)
+    # Traverse nodes when level is above leaves
+
+    # Compute number of virtual elements before this level to skip when computing the memory index
+    virtual_nodes_level = bvh.tree.virtual_leaves >> (bvh.tree.levels - (level - 1))
+    virtual_nodes_before = 2 * virtual_nodes_level - count_ones(virtual_nodes_level)
+
+    block_size = options.block_size
+    backend = get_backend(src)
+
+    kernel! = _traverse_rays_nodes_gpu!(backend, block_size)
+    kernel!(
+        bvh.tree, bvh.nodes, points, directions,
+        src, dst, num_src, dst_offsets, 
+        virtual_nodes_before,
+    )
+
```

Comment:
Like in traverse_*_gpu, we need to return the number of elements written to dst. Leave two newlines between functions